addons/delivery/models/delivery_carrier.py

```diff
@@ -63,19 +78,27 @@ def get_price(self):
         if order_id:
             # FIXME: temporary hack until we refactor the delivery API in master
 
-            if self.delivery_type != 'grid':
+            order = SaleOrder.browse(order_id)
+            if self.delivery_type not in ['fixed', 'base_on_rule']:
                 try:
-                    order = SaleOrder.browse(order_id)
                     self.price = self.get_shipping_price_from_so(order)[0]
                     self.available = True
                 except UserError as e:
+                    # no suitable delivery method found, probably configuration error
+                    _logger.info("Carrier %s: %s, not found", self.name, e.name)
+                    self.price = 0.0
+            else:
+                carrier = self.carrier_get(order.partner_shipping_id)
+                if carrier:
+                    try:
+                        self.price = carrier.get_price_available(order)
+                        self.available = True
+                    except UserError, e:
```

Comment:
All this part is strange. We need to look at it more closely when I get a closer grasp on the overall changes.